superbench/benchmarks/micro_benchmarks/gpu_copy_performance/gpu_copy.cu

```diff
@@ -104,51 +118,61 @@ struct Opts {
 
     // Whether bidirectional transfer is enabled.
     bool bidirectional_enabled = false;
+
+    // Whether check data after copy.
+    bool check_data = false;
 };
 
 // Print usage of this program.
 void PrintUsage() {
     printf("Usage: gpu_copy "
            "--size <size> "
+           "--num_warm_up <num_warm_up> "
            "--num_loops <num_loops> "
            "[--sm_copy] "
            "[--dma_copy] "
            "[--htod] "
            "[--dtoh] "
            "[--dtod] "
-           "[--bidirectional]\n");
+           "[--bidirectional] "
+           "[--check_data]\n");
 }
 
 // Parse options of this program.
 int ParseOpts(int argc, char **argv, Opts *opts) {
     enum class OptIdx {
         kSize,
-        kNumIters,
+        kNumWarmUp,
+        kNumLoops,
         kEnableSmCopy,
         kEnableDmaCopy,
         kEnableHToD,
         kEnableDToH,
         kEnableDToD,
-        kEnableBidirectional
+        kEnableBidirectional,
+        kEnableCheckData
     };
     const struct option options[] = {
         {"size", required_argument, nullptr, static_cast<int>(OptIdx::kSize)},
-        {"num_loops", required_argument, nullptr, static_cast<int>(OptIdx::kNumIters)},
+        {"num_warm_up", required_argument, nullptr, static_cast<int>(OptIdx::kNumWarmUp)},
+        {"num_loops", required_argument, nullptr, static_cast<int>(OptIdx::kNumLoops)},
         {"sm_copy", no_argument, nullptr, static_cast<int>(OptIdx::kEnableSmCopy)},
         {"dma_copy", no_argument, nullptr, static_cast<int>(OptIdx::kEnableDmaCopy)},
         {"htod", no_argument, nullptr, static_cast<int>(OptIdx::kEnableHToD)},
         {"dtoh", no_argument, nullptr, static_cast<int>(OptIdx::kEnableDToH)},
         {"dtod", no_argument, nullptr, static_cast<int>(OptIdx::kEnableDToD)},
-        {"bidirectional", no_argument, nullptr, static_cast<int>(OptIdx::kEnableBidirectional)}};
+        {"bidirectional", no_argument, nullptr, static_cast<int>(OptIdx::kEnableBidirectional)},
+        {"check_data", no_argument, nullptr, static_cast<int>(OptIdx::kEnableCheckData)}};
     int getopt_ret = 0;
     int opt_idx = 0;
     bool size_specified = false;
+    bool num_warm_up_specified = false;
     bool num_loops_specified = false;
     bool parse_err = false;
     while (true) {
         getopt_ret = getopt_long(argc, argv, "", options, &opt_idx);
         if (getopt_ret == -1) {
-            if (!size_specified || !num_loops_specified) {
+            if (!size_specified || !num_warm_up_specified || !num_loops_specified) {
```

Comment:
I am curious why we need this condition?

Reply:
Warm up is necessary to get stable performance number, so we made it a required option.